packages/react-internal/src/components/Layer/Layer.base.tsx

```diff
@@ -90,31 +90,21 @@ export const LayerBase: React.FunctionComponent<ILayerProps> = React.forwardRef<
     };
 
     React.useLayoutEffect(() => {
-      // During the initial render and any hostId updates:
-      //
+      createLayerElement();
```

Comment:
Is this a case where the logic didn't get translated quite correctly compared to the non-FC implementation?

Reply:
yes. the bug here is mainly this logic
```
      if (hostId) {
        registerLayer(hostId, createLayerElement);
      }
      // The user didn't provide a hostId, create a new layer element.
      else {
        createLayerElement();
      }
```

We should always call `createLayerElement` otherwise it won't be called. `LayerHost` only `notifyHostChanged` (calls  `createLayerElement`) when it's mounted. I believe the "notification" logic between `Layer` and  `LayerHost` is mainly to make sure the order of mounting these 2 components do not matter